modules/Core/queries/users.php

```diff
@@ -1,18 +1,30 @@
 <?php
-// Searchable user list
-if (!$user->isLoggedIn()) {
-    die(json_encode(['error' => 'Unauthenticated']));
+$validate = Validate::check($_GET, [
+    'search' => [
+        Validate::RATE_LIMIT => 60,
+    ],
+]);
+
+if (!$validate->passed()) {
+    die(json_encode(['error' => 'Please wait a moment before searching again']));
 }
 
-if (!isset($_GET['search']) || strlen($_GET['search']) < 3) {
-    die(json_encode(['error' => 'Please enter a search query of at least 3 characters']));
+// Searchable user list
+if (!isset($_GET['search']) || strlen($_GET['search']) < 2) {
+    die(json_encode(['error' => 'Please enter a search query of at least 2 characters']));
 }
 
 $query = '%' . $_GET['search'] . '%';
 
-$users = DB::getInstance()->query('SELECT id, username, nickname FROM nl2_users WHERE username LIKE ? OR nickname LIKE ?', [
+$users = DB::getInstance()->query('SELECT id, username, nickname, gravatar, email, has_avatar, avatar_updated FROM nl2_users WHERE username LIKE ? OR nickname LIKE ? LIMIT 5', [
```

Comment:
Not sure if it should be limited to 5 what else used this queries page to get users private message suggestions? should maybe be get param instead &limit=5 if isset